src/assist/reflexion_agent.py

```diff
@@ -184,14 +245,19 @@ def execute_node(state: ReflexionState) -> ReflexionState:
                 "A replan may be needed if this step was crucial."
             )
             state["plan_check_needed"] = True
+        
         res = StepResolution(
             action=step.action,
             objective=step.objective,
             resolution=resolution,
         )
-        state["history"] = state["history"] + [res]
-        state["step_index"] = step_index + 1
-        return state
+        
+        # Update state
+        new_state = state.copy()
+        new_state["messages"] = current_messages
+        new_state["history"] = state["history"] + [res]
+        new_state["step_index"] = step_index + 1
```

Comment:
Why this? Seems like it's outside the scope of the request. I'm ok with it since it seems like a positive change, but want to make sure.